pkg/runner/cluster_k8s.go

```diff
@@ -293,6 +295,146 @@ func (*ClusterK8sRunner) ID() string {
+
+	client := c.pool.Acquire()
+	defer c.pool.Release(client)
+
+	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
+		LabelSelector: "app=redis",
+	})
+	if err != nil {
+		redisCheck.Message = err.Error()
+		return
+	}
+	if len(pods.Items) != 1 {
+		redisCheck.Message = fmt.Sprintf("expected 1 redis pod. found %d.", len(pods.Items))
+		return
+	}
+
+	pod := pods.Items[0]
+	if pod.Status.Phase != "Running" {
+		return
+	}
+
+	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusOK, Message: "redis service is running"}
+	return
+}
+
+func (c *ClusterK8sRunner) healthcheckSidecar() (sidecarCheck api.HealthcheckItem) {
+	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusFailed, Message: "sidecar service is not reachable"}
+
+	// get number of worker nodes
+	client := c.pool.Acquire()
+	defer c.pool.Release(client)
+
+	res, err := client.CoreV1().Nodes().List(metav1.ListOptions{
+		LabelSelector: "kubernetes.io/role=node",
+	})
+	if err != nil {
+		sidecarCheck.Message = err.Error()
+		return
+	}
+
+	nodes := len(res.Items)
+
+	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
+		LabelSelector: "name=testground-sidecar",
+	})
+	if err != nil {
+		sidecarCheck.Message = err.Error()
+		return
+	}
+	if len(pods.Items) != nodes {
+		sidecarCheck.Message = fmt.Sprintf("expected %d sidecar pods. found %d.", nodes, len(pods.Items))
+		return
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != "Running" {
+			return
+		}
+	}
+
+	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusOK, Message: "sidecar service is running"}
+	return
+}
+
+func (c *ClusterK8sRunner) Healthcheck(fix bool, engine api.Engine, writer io.Writer) (*api.HealthcheckReport, error) {
+
+	// TODO: initialize the runner correctly.
+	// When the runner is unitialized, that is, when it has a nil pointer where one is needed.
+	// I considered changing DoHealthCheck to pass an input parameter, such as we do for DoRun and
+	// DoCollectOutputs, but there are already other TODO's in this file that makes me think
+	// initRunner needs to be overhauled. So until such time that there is better initialization,
+	// handle, I'll handle uninitialized runners by sending an empty report here.
+	// What this means is if the first thing you do is run the health check, you will get an empty
+	// report, but for subsequent times, the report is real.
+	report := api.HealthcheckReport{}
+
+	if c.pool != nil {
+		report.Checks = []api.HealthcheckItem{
+			c.healthcheckK8s(),
+			c.healthcheckEFS(),
+			c.healthcheckRedis(),
+			c.healthcheckSidecar(),
+		}
+
+		if fix {
+			fakeFixes := []api.HealthcheckItem{}
+			for _, chk := range report.Checks {
+				if chk.Status != api.HealthcheckStatusOK {
+					fakeFixes = append(fakeFixes, api.HealthcheckItem{
+						Name:    chk.Name,
+						Status:  chk.Status,
```

Comment:
`Status` should be "Omitted" here. We haven't performed the fix, it can't have failed.